azkaban-common/src/main/java/azkaban/utils/Emailer.java

```diff
@@ -46,8 +48,9 @@
   private final String tls;
   private boolean testMode = false;
 
-  public Emailer(final Props props) {
+  public Emailer(final Props props, final CommonMetrics commonMetrics) {
     super(props);
+    this.commonMetrics = commonMetrics;
```

Comment:
I just realized I directly use the class `EmailMessage`, rather than Emailer. It is safe to have Metrics in the constructor.